zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/ProcessInstanceCreationCreatedApplierV2.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.state.appliers;
+
+import io.camunda.zeebe.engine.processing.deployment.model.element.ExecutableFlowNode;
+import io.camunda.zeebe.engine.state.TypedEventApplier;
+import io.camunda.zeebe.engine.state.immutable.ProcessState;
+import io.camunda.zeebe.engine.state.instance.ElementInstance;
+import io.camunda.zeebe.engine.state.mutable.MutableElementInstanceState;
+import io.camunda.zeebe.engine.state.mutable.MutableProcessState;
+import io.camunda.zeebe.protocol.impl.record.value.processinstance.ProcessInstanceCreationRecord;
+import io.camunda.zeebe.protocol.record.intent.ProcessInstanceCreationIntent;
+import io.camunda.zeebe.protocol.record.value.BpmnElementType;
+import java.util.List;
+import org.agrona.DirectBuffer;
+
+final class ProcessInstanceCreationCreatedApplierV2
```

Comment:
💭 When creating new applier versions, I recommend splitting up the commits
- 1 commit to copy the V1 applier to a new V2 applier
- 1 commit adjusting the V2 applier so the difference with V1 is clear.